### PR TITLE
Public-sans - POAM: September '24

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

POAM updates for September 2024

>[!WARNING]
>We received deprecation warnings for dependencies that are no longer supported. They are coming from USWDS compile and will be resolved in uswds/uswds-compile#122.

> [!IMPORTANT]
> This PR caught a Federalist build issue. The issue appears unrelated to these changes but was caught due to generating a new `gemfile.lock`.
>
> The federalist pages team is investigating. Additional details in this [slack thread](https://gsa-tts.slack.com/archives/C04823NMA/p1725995573366869) (🔒). 
>
> In the meantime I've downgraded ruby.

## Related issue

uswds/uswds-team#390
Resolves https://github.com/uswds/public-sans/security/dependabot/84
Resolves https://github.com/uswds/public-sans/security/dependabot/83
Resolves https://github.com/uswds/public-sans/security/dependabot/74
Resolves https://github.com/uswds/public-sans/security/dependabot/81
Resolves https://github.com/uswds/public-sans/security/dependabot/82

## Preview link

[Preview link →](https://federalist-514d1714-e827-44f8-a29f-3ec1837bf619.sites.pages.cloud.gov/preview/uswds/public-sans/cm-POAM-september-2024/)

## Major changes

- Ruby downgraded from `3.3.4` to `3.2.5` to resolve Cloud Pages build error
- Updated USWDS to `3.9.0`

## Dependency updates

**Before:**

```node
11 vulnerabilities (6 moderate, 5 high)
```

**After**

```node
found 0 vulnerabilities
```

### Node package updates

| Dependency name | Old version | New version |
| --- | --- | --- |
| @axe-core/cl | ^4.9.1 | ^4.10.0 |
| @uswds/uswds | 3.8.1 | 3.8.2 |
| gulp | ^4.0.2 | ^5.0.0 |
| postcss | ^8.4.41 | ^8.4.45 |
| sass-embedded | ^1.77.8 | ^1.78.0 |

### Gem updates:

| Dependency name | Old version | New version |
| --- | --- | --- |
| @uswds/uswds | 3.8.2 | 3.9.0 |
| concurrent-ruby | 1.3.3 | 1.3.4 |
| google-protobuf | 4.28.0 | 4.28.2 |
| i18n | 1.14.5 | 1.14.6 |
| jekyll | 4.3.3 | 4.3.4 |
| rexml | 3.3.4 | 3.3.8 |
| rouge | 4.3.0 | 4.4.0 |
| postcss | ^8.4.45 | ^8.4.47 |
| sass-embedded | 1.78.0 | 1.79.4 |
| strscan | 3.1.0 | — |
| unicode-display_width | 2.5.0 | 2.6.0 |
| webrick | 1.8.1 | 1.8.2 |

## Testing and review

Gulp commands run without error
1. `npm run start`
2. `npm run serve`
3. `npm run test:a11y` (while localhost is being served from the serve script)
4. Confirm no font regressions in Public Sans fonts due to Gulp update

### CODE REVIEW

**Review by mejiaj — APPROVED** [review]
@mahoneycm thanks for the notes in the description. Hope you don't mind, I've modified the `Important` alert to state the workaround (last sentence - _In the meantime I've downgraded ruby._

I've been able to successfully switch to Ruby 3.25 and do a clean install of both Node & Ruby dependencies without issues.

Tested using `npm run serve` and `npm start`.

**Comment by mahoneycm** [comment]
## October updates

**Dependency updates**

| Dependency name | Old version | New version |
| --- | --- | --- |
| @uswds/uswds | 3.8.1 | 3.8.2 |
| postcss | ^8.4.41 | ^8.4.45 |
| sass-embedded | ^1.77.8 | ^1.78.0 |

**Gem updates**

| Gem name | Old version | New version |
| --- | --- | --- |
| google-protobuf | 4.28.1 | 4.28.2 |
| rexml | 3.3.7 | 3.3.8 |
| sass-embedded | 1.78.0 | 1.79.4 |
| webrick | 1.8.1 | 1.8.2 |

**Comment by mahoneycm** [comment]
>[!IMPORTANT]
>Converting to draft while we review compile POAM PR

File: package.json [hunk 1]
```diff
@@ -37,15 +37,15 @@
   },
   "dependencies": {
     "@uswds/compile": "1.2.0",
-    "@uswds/uswds": "3.8.1",
-    "gulp": "^4.0.2",
+    "@uswds/uswds": "3.9.0",
+    "gulp": "^5.0.0",
```
Comment (mejiaj) [inline package.json:41]: > [!TIP]
> This major Gulp update has caused issues with fonts and images. I tested `npm run copy-webfonts` and didn't see any issues.

**Review by mejiaj — APPROVED** [review]
LGTM. 

I've added a note about Gulp 5 and fonts/images, but didn't see any regressions at first glance.

File: package.json (hunk 1 above)
Comment (mahoneycm) [reply package.json:41]: I took a look at your work in uswds/uswds-compile#96 to make sure the issue that was captured and resolved there wasn't going to be an issue here! I didn't find any issues either 👍

File: gulpfile.js [hunk 2]
```diff
@@ -13,8 +13,9 @@ const WEBFONTS_SRC = "./fonts/webfonts";
 const WEBFONTS_DEST = "./site/assets/fonts";
 
 function copyWebfonts() {
-  return src(`${WEBFONTS_SRC}/**/**`)
-    .pipe(dest(WEBFONTS_DEST));
+  return src(`${WEBFONTS_SRC}/**/**`, {
+    encoding: false,
+  }).pipe(dest(WEBFONTS_DEST));
```
Comment (thisisdano) [inline gulpfile.js:18]: Added to address the issue fixed in https://github.com/uswds/uswds-compile/pull/96 from a solution noted in https://github.com/gulpjs/gulp/issues/2803

**Review by thisisdano — APPROVED** [review]
LGTM

**Pull request merged**